### DISCUSSION TITLE
DuckDB storage backward compatibility

### DISCUSSION THREAD

**Discussion by lhoestq** [duckdb/duckdb#8045] [General]
Hi ! Currently the DuckDB storage format changes almost at every release (as noted at https://duckdb.org/internals/storage).
Therefore one always needs to convert old formats to the new ones using EXPORT/IMPORT.

Is some sort of backward compatibility planned in the near future ?
This would be useful since we're starting to use DuckDB quite extensively at Hugging Face

**Comment by Mytherin** [comment]
We are planning to stabilize the storage starting with version 1.0, which is planned for later this year. After version 1.0 the guarantee we plan to make is backwards compatibility, i.e. any version after can read database files generated by any prior version (i.e. v1.1 will be able to read files generated by v1.0, but not necessarily vice versa).
